common.yml

```diff
@@ -904,6 +912,91 @@ functions:
           fi;
           ./bin/mongo $MONGO_ARGS --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:${mongod_port}"); return true} catch(e){return false}}, "timed out connection")'
 
+  "add-aws-auth-variables-to-file":
+    - command: shell.exec
+      type: test
+      params:
+        silent: true
+        working_dir: src/github.com/mongodb/mongo-tools/common/lib
+        script: |
+          set -x
+          set -v
+          set -e
+
+          cat <<EOF > $(pwd)/aws_e2e_setup.json
+          {
+            "iam_auth_assume_aws_account": "${iam_auth_assume_aws_account}",
+            "iam_auth_assume_aws_secret_access_key": "${iam_auth_assume_aws_secret_access_key}",
+            "iam_auth_assume_role_name": "${iam_auth_assume_role_name}"
+          }
+          EOF
+
+  "setup-aws-auth-test-with-assume-role-credentials":
+    - command: shell.exec
+      type: test
+      params:
+        silent: true
+        working_dir: src/github.com/mongodb/mongo-tools/common/lib
+        script: |
+          set -x
+          set -v
+          set -e
+
+          # The aws_e2e_assume_role script requires python3 with boto3.
+          # Set up or use an existing python virtualenv for boto3.
+          venv='venv'
+          venvNonWindows () { . "$venv"/bin/activate; python3 -m pip install boto3; }
+          venvWindows () { . "$venv"/Scripts/activate; pip install boto3; }
```

Comment:
I copied this section from sqlproxy's approach to python venvs for Windows vs. Non-Windows. We were actually installing `boto3` globally on Windows machines in the mtc script.